models/double_entry_transactions/int_quickbooks__invoice_double_entry.sql

```diff
@@ -104,9 +104,19 @@ invoice_join as (
                 end as amount,
 
         {% if var('using_invoice_bundle', True) %}
-        coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id) as account_id,
+        case when invoice_lines.detail_type is not null then invoice_lines.detail_type
+            when coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id) is not null then 'SalesItemLineDetail'
+            when invoice_lines.discount_account_id is not null then 'DiscountLineDetail'
+            when coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id, invoice_lines.discount_account_id) is null then 'DescriptionOnly'
+        end as invoice_line_transaction_type,
+        coalesce(invoice_lines.account_id, items.parent_income_account_id, items.income_account_id, bundle_income_accounts.account_id, invoice_lines.discount_account_id) as account_id,
         {% else %}
-        coalesce(invoice_lines.account_id, items.income_account_id) as account_id,
+        case when invoice_lines.detail_type is not null then invoice_lines.detail_type
```

Comment:
Quick clarifying question: The when clause was changed because detail_type will have the SubTotalLineDetail? What does DescriptionOnly refer to, an invoice line item ro wwhere there's an amount but no id associated with it?

Reply:
Exactly!